library/Cube/IcingaDb/IcingaDbCube.php

```diff
@@ -87,6 +89,58 @@ public function addDimensionByName($name)
         return $this;
     }
 
+    /**
+     * Fetch the host variable dimensions
+     *
+     * @return array
+     */
+    public function fetchHostVariableDimensions(): array
+    {
+        $query = Host::on($this->getDb())
+            ->with('customvar_flat')
+            ->columns('customvar_flat.flatname')
+            ->orderBy('customvar_flat.flatname');
+
+        $this->applyRestrictions($query);
+
+        $query->getSelectBase()->groupBy('flatname');
+
+        $dimensions = [];
+        foreach ($query as $row) {
+            // Replaces array index notations with [*] to get results for arbitrary indexes
+            $name = preg_replace('/\\[\d+](?=\\.|$)/', '[*]', $row->customvar_flat->flatname);
```

Comment:
Please lower the name, custom vars are handled case-insensitive by Icinga DB Web.